libs/langchain/langchain/document_loaders/web_base.py

```diff
@@ -208,7 +211,11 @@ def _scrape(self, url: str, parser: Union[str, None] = None) -> Any:
         html_doc = self.session.get(url, **self.requests_kwargs)
         if self.raise_for_status:
             html_doc.raise_for_status()
-        html_doc.encoding = html_doc.apparent_encoding
+
+        if self.encoding is not None:
+            html_doc.encoding = self.encoding
+        elif self.autoset_encoding:
+            html_doc.encoding = html_doc.apparent_encoding
```

Comment:
out of curiosity, what happens when html_doc.encoding isn't set?

Reply:
According to requests it's "an educated guess" [source](https://docs.python-requests.org/en/latest/user/quickstart/#response-content]. Perhaps based on the [content type](https://stackoverflow.com/a/52615216/8207)?

In practice I observe that it's usually utf-8, which seems reasonable